src/pages/resources/recipes/file-operation.md

```diff
@@ -0,0 +1,317 @@
+# File operations
+
+To perform any file operations including read, write, create, and delete, UXP provides a couple of options. But before we look at the APIs, let's get acquainted with a few concepts.
+
+## Concepts
+
+### Sandbox and other locations
+UXP, by default, only allows access to certain locations in the user's file system. These locations are referred to as the sandbox. 
+
+<InlineAlert variant="info" slots="header, text1, text2"/>
+
+Scripts and plugins
+
+**In plugins**, a sandbox is typically the plugin's folder, plus a temporary and data folder. The plugin's location is read-only. Whereas, the temporary and data folders are provided to you to store transitory information. Note that the data stored here can get accidentally erased and should not be considered permanent.
+
+**In scripts**, lorem ipsum. // TODO What about sandbox locations in scripts?
+
+
+However, we understand that there are circumstances when you would like to have access to other locations as well. Accessing such locations, too, is possible in UXP but you will need to seek permission first.
+
+
+### Manifest permission
+For file system access you require permission for the `localFileSystem` module.
+
+<InlineAlert variant="info" slots="header, text1, text2"/>
+
+Scripts and plugins
+
+**In scripts**, these permissions are fixed to '' and you cannot alter them //(TODO Add link to details) and ignore the manifest details below. 
+
+**In plugins**, you should seek the permission for `localFileSystem` in the your manifest.<br></br> IMPORTANT: Be sure you know about the manifest permission module before you proceed) //TODO insert link to manifest -> network module.
+
+
+Lets understand the manifest settings a bit more in detail.
+```json
+/* manifest.json */
+{  
+    "requiredPermissions": {
+        "localFileSystem": "plugin"
+    }
+}
+```
+Allowed values for `localFileSystem` are:
+- `plugin`: Default value. Provides access to the sandbox.
+- `request`: Let's you request the user to select a folder location of their choice.
+- `fullAccess`: Provides full access to the user file system
+//TODO
+Q: Does fullAccess file system include system folders?
```

Comment:
@kareemmyk

Reply:
Let me know in case you need more details

Reply:
What do you mean by 'local storage?
How about localFilesystem API?